bin/pkg/lock.ml

```diff
@@ -46,30 +107,25 @@ let solve_lock_dir
       ~unset_solver_vars_from_context:
         (unset_solver_vars_of_workspace workspace ~lock_dir_path)
   in
-  let overlay =
-    Console.Status_line.add_overlay (Constant (Pp.text "Solving for Build Plan"))
-  in
   let* repos =
-    repositories_of_workspace workspace
-    |> get_repos ~repositories:(repositories_of_lock_dir workspace ~lock_dir_path)
+    let repo_map = repositories_of_workspace workspace in
+    let repo_names = Dune_pkg.Pkg_workspace.Repository.Name.Map.keys repo_map in
```

Comment:
I think we should make sure to sort them to make sure that the order is always consistent.